### PR TITLE
Channel request

### PR DESCRIPTION
**Pull request by narangodapasan**
add crud operation for channeling

### CODE REVIEW

File: apps/api/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.env
```
Comment (chxru) [inline apps/api/.gitignore]: Move .env config to root dir

File: apps/api/src/app/app.module.ts [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import { Module } from '@nestjs/common';
-
+import { MongooseModule } from '@nestjs/mongoose';
 import { AppController } from './app.controller';
 import { AppService } from './app.service';
+import { ChannelingModule } from './channeling/channeling.module';
+require('dotenv').config();
```
Comment (chxru) [inline apps/api/src/app/app.module.ts]: Use nestjs's config module
https://docs.nestjs.com/techniques/configuration

File: apps/api/src/app/channeling/channeling.controller.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
```
Comment (chxru) [inline apps/api/src/app/channeling/channeling.controller.ts]: Unused import

File: apps/api/src/app/app.module.ts [hunk 4]
```diff
@@ -1,11 +1,13 @@
 import { Module } from '@nestjs/common';
-
+import { MongooseModule } from '@nestjs/mongoose';
```
Comment (chxru) [inline apps/api/src/app/app.module.ts:2]: Package is missing in package.json

File: apps/api/src/app/channeling/channeling.module.ts [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { Module } from '@nestjs/common';
+import { ChannelingService } from './channeling.service';
+import { ChannelingController } from './channeling.controller';
+import { MongooseModule } from '@nestjs/mongoose';
```
Comment (chxru) [inline apps/api/src/app/channeling/channeling.module.ts:4]: Package is missing in package.json

File: apps/api/src/app/channeling/channeling.service.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { Channeling } from './interfaces/channeling.interface';
+import { Model } from 'mongoose';
+import { InjectModel } from '@nestjs/mongoose';
+import { CreateChannelingDto } from './dto/create-channeling.dto';
+import { UpdateChannelingDto } from './dto/update-channeling.dto';
```
Comment (chxru) [inline apps/api/src/app/channeling/channeling.service.ts]: Unused import

File: apps/api/src/app/channeling/channeling.service.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@nestjs/common';
+import { Channeling } from './interfaces/channeling.interface';
+import { Model } from 'mongoose';
```
Comment (chxru) [inline apps/api/src/app/channeling/channeling.service.ts:3]: Package is missing in package.json

File: apps/api/src/app/channeling/entities/channeling.entity.ts [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import * as mongoose from 'mongoose';
```
Comment (chxru) [inline apps/api/src/app/channeling/entities/channeling.entity.ts]: Package is missing in package.json

**Review by chxru — CHANGES_REQUESTED** [review]

File: apps/api/.gitignore (hunk 1 above)
Comment (narangodapasan) [reply apps/api/.gitignore]: moved to root dir

File: apps/api/src/app/app.module.ts (hunk 4 above)
Comment (narangodapasan) [reply apps/api/src/app/app.module.ts:2]: added missing package

File: apps/api/src/app/app.module.ts (hunk 2 above)
Comment (narangodapasan) [reply apps/api/src/app/app.module.ts]: changed the code structure, check this

File: apps/api/src/app/channeling/channeling.service.ts (hunk 7 above)
Comment (narangodapasan) [reply apps/api/src/app/channeling/channeling.service.ts:3]: mongoose package added

File: apps/api/src/app/channeling/channeling.service.ts (hunk 6 above)
Comment (narangodapasan) [reply apps/api/src/app/channeling/channeling.service.ts]: removed import

File: apps/api/src/app/channeling/channeling.controller.ts (hunk 3 above)
Comment (narangodapasan) [reply apps/api/src/app/channeling/channeling.controller.ts]: removed unused import

**Review by chxru — CHANGES_REQUESTED** [review]
package-lock.json should not exists. Use yarn for all dependency installs

File: apps/api/src/app/channeling/channeling.service.ts (hunk 6 above)
Comment (chxru) [reply apps/api/src/app/channeling/channeling.service.ts]: @narangodaNPP UpdateChannelingDto is still unused

**Review by chxru — CHANGES_REQUESTED** [review]

**Review by chxru — APPROVED** [review]

**Pull request merged**